File: IT/1/Oggetti/Esercizi/C++/esercizio27.cpp
/* file esercizio27.cpp */

#include <iostream>
#include <iomanip>
#include <cmath>

using std::cout;
using std::endl;
using std::string;
using std::setprecision;
using std::ostream;

/* la classe Circle rappresenta una circonferenza */

class Circle {
	/* ogni circonferenza viene definita dal suo raggio */
	private:
		double r;

	public:
		Circle(double);
		double area();
};

Circle::Circle(double r) {
	this->r = r;
}

/* area del cerchio = Pi greco * raggio al quadrato */
double Circle::area() {
	return M_PI * this->r * this->r;
}

/* la classe Cilinder rappresenta un cilindro tridimensionale */

class Cilinder {
	/* ogni cilindro viene definito da altezza e circonferenza alla base */
	private:
		Circle base;
		double h;

	public:
		Cilinder(Circle&, double);
		double volume();
};

Cilinder::Cilinder(Circle& base, double h) : base(base) {
	this->h = h;
}

/* volume del cilindro: 3.14 * r^2 * h */
double Cilinder::volume() {
	return this->base.area() * this->h;
}

int main()
{
	Circle circle(2.0);

	cout << "Area circonferenza alla base: " << setprecision(4) << circle.area() << " centimetri quadrati." << endl;

	Cilinder solid(circle, 3.0);

	cout << "Volume cilindro: " << setprecision(4) << solid.volume() << " centimetri cubi." << endl;
}

File: IT/1/Oggetti/Esercizi/Python/esercizio27.py
# file esercizio27.py

from __future__ import print_function
from math import pi

# la classe Circle rappresenta una circonferenza

class Circle(object):
	# ogni circonferenza viene definita dal suo raggio
	def __init__(self, r):
		self.__r = r

	# area del cerchio = Pi greco * raggio al quadrato
	def area(self):
		return pi * self.__r * self.__r

# la classe Cilinder rappresenta un cilindro tridimensionale

class Cilinder(object):
	# ogni cilindro viene definito da altezza e circonferenza alla base
	def __init__(self, base, h):
		self.__base = base
		self.__h = h

	# volume del cilindro: 3.14 * r^2 * h
	def volume(self):
		return self.__base.area() * self.__h

circle = Circle(r=2.0)

print("Area circonferenza alla base: {:.2f} centimetri quadrati.".format(circle.area()))

solid = Cilinder(base=circle, h=3.0)

print("Volume cilindro: {:.2f} centimetri cubi.".format(solid.volume())) 

